src/Shared/Traits.cs

```diff
@@ -145,6 +145,8 @@ internal class EscapeHatches
         /// </summary>
         public readonly bool AlwaysEvaluateDangerousGlobs = Environment.GetEnvironmentVariable("MSBuildAlwaysEvaluateDangerousGlobs") == "1";
 
+        public readonly bool DoNotOptimizeConsoleLogger = Environment.GetEnvironmentVariable("MSBUILDDONOOPTIMISECONSOLELOGGER") == "1";
```

Comment:
nit: Typo in the env var name and inconsistent S/Z.
```suggestion
        public readonly bool DoNotOptimizeConsoleLogger = Environment.GetEnvironmentVariable("MSBUILDDONOTOPTIMIZECONSOLELOGGER") == "1";
```

Reply:
This Trait has been remove. This change will be under ChangeWaves only.